### ISSUE TITLE
eng_python() seems to ignore a single line of code that does not generate output

### ISSUE THREAD

**Issue by yihui** [rstudio/reticulate#126]
Minimal reprex:

````
```{r}
knitr::knit_engines$set(python = reticulate::eng_python)
```

```{python}
x = "hi there"
print(x)
```

```{python}
y = x.split(" ")  # no output
```

```{python}
print(y)
```
````

Output (note that the second python code chunk is missing):

![image](https://user-images.githubusercontent.com/163582/32962024-73e4e0a4-cb90-11e7-872a-4f88a77c7c56.png)

**Comment by jjallaire** [comment]
@kevinushey Let's see if we can get a fix in for this one (realize you aren't focused on the python engine right now but I think this might be a deal-breaker for knitr making reticulate the default)

**Comment by jjallaire** [comment]
I had to do a patch release of reticulate for another reason and as a result this fix is now on CRAN in v1.3.1: https://cran.r-project.org/web/packages/reticulate/index.html

**Comment by yihui** [comment]
Excellent! I'll prepare a CRAN release of knitr soon.
